app/services/ctdl_query.rb

```diff
@@ -0,0 +1,492 @@
+      return conditions << build(term_group)
+    end
+
+    node.map do |key, value|
+      next if key == 'search:operator'
+
+      build_condition(key, value)
+    end.compact
+  end
+
+  def build_fts_condition(config, key, term)
+    return table[key].not_eq(nil) if term == ANY_VALUE
+
+    if term.is_a?(Array)
+      conditions = term.map { |t| build_fts_condition(config, key, t) }
+      return combine_conditions(conditions, :or)
+    end
+
+    term = term.fetch('search:value') if term.is_a?(Hash)
+    quoted_config = Arel::Nodes.build_quoted(config)
+
+    translated_column = Arel::Nodes::NamedFunction.new(
+      'translate',
+      [
+        table[key],
+        Arel::Nodes.build_quoted('/.'),
+        Arel::Nodes.build_quoted(' ')
+      ]
+    )
+
+    translated_term = Arel::Nodes::NamedFunction.new(
+      'translate',
+      [
+        Arel::Nodes.build_quoted(term),
+        Arel::Nodes.build_quoted('/.'),
+        Arel::Nodes.build_quoted(' ')
+      ]
+    )
+
+    column_vector = Arel::Nodes::NamedFunction.new(
+      'to_tsvector',
+      [quoted_config, translated_column]
+    )
+
+    query_vector = Arel::Nodes::NamedFunction.new(
+      'plainto_tsquery',
+      [quoted_config, translated_term]
+    )
+
+    fts_columns << table[key]
+
+    fts_ranks << Arel::Nodes::NamedFunction.new(
+      'ts_rank',
+      [column_vector, query_vector]
+    )
+
+    Arel::Nodes::InfixOperation.new('@@', column_vector, query_vector)
+  end
+
+  def build_fts_conditions(key, value)
+    conditions = value.items.map do |item|
+      if item.is_a?(Hash)
+        conditions = item.map do |locale, term|
+          name = "#{key}_#{locale.tr('-', '_').downcase}"
+          column = columns_hash[name]
+          next IMPOSSIBLE_CONDITION unless column
+
+          config =
+            if locale.starts_with?('es')
+              'spanish'
+            elsif locale.starts_with?('fr')
+              'french'
+            else
+              'english'
+            end
+
+          build_fts_condition(config, name, term)
+        end
+      elsif item.is_a?(SearchValue)
+        build_fts_condition('english', key, item.items)
+      elsif item.is_a?(String)
+        build_fts_condition('english', key, item)
+      else
+        raise "FTS condition should be either an object or a string, `#{item}` is neither"
+      end
+    end.flatten
+
+    combine_conditions(conditions, value.operator)
+  end
+
+  def build_id_condition(key, values)
+    conditions = values.map do |value|
+      if full_id_value?(key, value)
+        table[key].eq(value)
+      else
+        table[key].matches("%#{value}%")
+      end
+    end
+
+    combine_conditions(conditions, :or)
+  end
+
+  def build_like_condition(key, values, match_type)
+    conditions = values.map do |value|
+      value =
+        case match_type
+        when 'search:endsWith' then "%#{value}"
+        when 'search:exactMatch' then "%#{value}%"
```

Comment:
@excelsior Isn't it backwards?

Seems like `exactMatch` should have no `%` signs.

Reply:
I consulted with Nate, and he clarified that `exact` in this case means `not approximate`, i.e. when this option is provided, the condition should match the substring withing a string without fuzzy search techniques.